### DISCUSSION TITLE
Reverb??

### DISCUSSION THREAD

**Discussion by ErichHeinemann** [marcel-licence/esp32_basic_synth#36] [Ideas]
Hi Marcel,
I have tried to add the Reverb from the Sampler to the Synth - it simply crashes.
I tried to replace the Delay with the Reverb  - it does nothing?

Before the ESP crashes, a beep with reverb was created but that´s all.
I used Your original code.

Any idea?
I added the needed Variable int the config.h
SAMPLE_BUFFER_SIZE=64

In Setup:
    Serial.printf("Initialize Reverb Module\n");
    Reverb_Setup();

In the loop, I did this:
Reverb_Process( &fl_sample, &fr_sample, SAMPLE_BUFFER_SIZE );

Any suggestions? I simply wanted to try the reverb as an alternative.
Would it be possible to add the BiFilter from the Drums too?

**Comment by marcel-licence** [comment]
Hi Erich,

yes it crashed because of accesing memory which shouldn't be used. The synth project doesn't use a sample buffer. You can try to use 
`#define SAMPLE_BUFFER_SIZE 1`
In addition to that please check if the memory allocation is successful. There are places using malloc and some memory is allocated as static RAM.

Yes I think that should work. Each feature will cost processing power limiting the max polyphony at the end. But it might work

**Comment by ErichHeinemann** [comment]
Hi Marcel


Keytracking:
    voice->velocity = vel * 1.25 - (note/500);// 0.25; /* just something to test */

Reverb copied from the Sampler: -> Sounds great with Unison!! Fat fatter!   And Delay works still fine too!
    Reverb_Process( &fl_sample, &fr_sample, 1);




Works perfekt!

**Comment by GillesLACAUD** [comment]
I will test the reverb too...good  idea.
How many poly voice did you have Erich....for me it is just 4 in poly mode...i will try to add a paraphonic mode also

**Comment by ErichHeinemann** [comment]
I was only using „Unison“-Mode with 3 fat voice - Sounding like the „JUMP“-Solo .. was enough for me.

Do You have any suggestions how Ground or Pin 2 has to be connected on die MIDI-In-Connector?
Normally it is not connected, but yesterday I had issues with the Novation Circuit, using its MIDI Out and connect to the MIDI-IN drove the ESP32-Synth to craziness. I had to add a CAP 2.2yF to ground to get proper MIDI-Signals back. Same Schematic worked fine with my Yamaha Keyboard. .. it is only the Novation Circuit doing wrong things.

Best Regards
Erich

**Comment by GillesLACAUD** [comment]
Maybe you must try to record what the Circuit send to his midi port (try this https://mountainutilities.eu/miditools).Could you add a schematics for all you mapping issues ?
